test/variational/test_variational_mixed.py

```diff
@@ -78,12 +78,17 @@ def vstate(request):
     return vs
 
 
-def test_n_samples_api(vstate):
+def test_n_samples_api(vstate, _mpi_size):
```

Comment:
that's a good idea.
However this whole file is marked with `pytestmark = common.skipif_mpi` so it's not run under MPI.
Since you made this test mpi-aware, you should delete that line and add the `@common.skipif_mpi` decorator to all the tests that should not run under MPI (I guess all but this one)